python_modules/dagster/dagster/_core/storage/asset_check_execution_record.py

```diff
@@ -38,6 +44,46 @@ class AssetCheckExecutionRecordStatus(enum.Enum):
 }
 
 
+@whitelist_for_serdes
+@record
+class AssetCheckPartitionStatusCacheValue:
+    """Serializable cache value stored in database for asset check partition status."""
+
+    latest_storage_id: int
+    partitions_def_id: Optional[str]
+    serialized_planned_subset: "SerializedPartitionsSubset"
+    serialized_succeeded_subset: "SerializedPartitionsSubset"
+    serialized_failed_subset: "SerializedPartitionsSubset"
+    # Map of partition key -> run_id for planned partitions (to resolve run status)
+    planned_partition_run_mapping: dict[str, str]
+
+    @classmethod
+    def from_partitions_def(
+        cls, partitions_def: "PartitionsDefinition"
+    ) -> "AssetCheckPartitionStatusCacheValue":
+        serialized_subset = SerializedPartitionsSubset.from_subset(
+            partitions_def.empty_subset(), partitions_def
+        )
+        return AssetCheckPartitionStatusCacheValue(
+            latest_storage_id=0,
+            partitions_def_id=partitions_def.get_serializable_unique_identifier(),
+            serialized_planned_subset=serialized_subset,
+            serialized_succeeded_subset=serialized_subset,
+            serialized_failed_subset=serialized_subset,
+            planned_partition_run_mapping={},
+        )
+
+
+@record
+class AssetCheckPartitionRecord:
+    """Record representing a single asset check partition with execution and run info."""
+
+    partition_key: str
+    last_execution_status: AssetCheckExecutionRecordStatus
```

Comment:
It would be really great if we could store the `AssetCheckExecutionResolvedStatus` as part of the record too - without it we have to do a bunch of additional computation at read time to determine if the check is in a failed state or not (see [here](https://github.com/dagster-io/dagster/blob/affa5128f1f58602438914365d563b69a8d57f95/python_modules/dagster/dagster/_core/definitions/asset_health/asset_check_health.py#L102-L145) for an example). this was a big reason why the old asset health page was so slow, so if we can avoid having to do this computation at all for partitioned checks that would be great